app/api/business/team_business.py

```diff
@@ -0,0 +1,289 @@
+from collections import namedtuple
+
+from flask_login import current_user
+
+from app.api.business.errors import NotFoundError, TeamError, ValidationError, UnauthorisedError
+from app.api.business.validators import TeamValidator
+from app.api.helpers import get_email_domain
+from app.api.services import (
+    audit_service,
+    audit_types,
+    team_member_permission_service,
+    team_member_service,
+    team_service,
+    users
+)
+from app.emails.teams import (
+    send_team_lead_notification_emails,
+    send_team_member_notification_emails,
+    send_request_access_email
+)
+from app.models import Team, TeamMember, TeamMemberPermission, permission_types
+from app.tasks import publish_tasks
+
+
+def create_team():
+    user = users.get(current_user.id)
+    created_teams = team_service.get_teams_for_user(user.id, 'created')
+    completed_teams = team_service.get_teams_for_user(user.id)
+
+    new_team = None
+    if len(completed_teams) == 0:
+        if len(created_teams) == 0:
+            team = team_service.save(
+                Team(
+                    name='',
+                    status='created',
+                    team_members=[
+                        TeamMember(
+                            user_id=user.id,
+                            is_team_lead=True
+                        )
+                    ]
+                )
+            )
+
+            audit_service.log_audit_event(
+                audit_type=audit_types.create_team,
+                db_object=team,
+                user=current_user.email_address
+            )
+
+            publish_tasks.team.delay(
+                publish_tasks.compress_team(team),
+                'created'
+            )
+
+            return get_team(team.id)
+
+        created_team = created_teams.pop()
+        return get_team(created_team.id)
+    else:
+        team = completed_teams[0]
+        raise TeamError('You can only be in one team. You\'re already a member of {}.'.format(team.name))
+
+
+def get_teams_overview():
+    teams_overview = {}
+    user = users.get(current_user.id)
+    completed_teams = team_service.get_teams_for_user(user.id)
+    email_domain = get_email_domain(user.email_address)
+
+    overview = team_service.get_teams_overview(user.id, email_domain)
+    teams_overview.update(overview=overview)
+
+    organisation = users.get_user_organisation(email_domain)
+    teams_overview.update(organisation=organisation)
+
+    show_create_team_button = True if len(completed_teams) == 0 else False
+    teams_overview.update(showCreateTeamButton=show_create_team_button)
```

Comment:
This feels like something that should be determined on the frontend.

Reply:
I've replaced this with a count of the completed teams.